lib/buffer.js

```diff
@@ -899,6 +899,10 @@ Buffer.prototype.compare = function compare(target,
 // - encoding - an optional encoding, relevant if val is a string
 // - dir - true for indexOf, false for lastIndexOf
 function bidirectionalIndexOf(buffer, val, byteOffset, encoding, dir) {
+  if (!isArrayBufferView(buffer)) {
+    throw new ERR_INVALID_ARG_TYPE('buffer', 'Buffer', buffer);
+  }
```

Comment:
Please use the validation function from `lib/internal/validators.js`.

```suggestion
  validateBuffer(buffer);
```

Reply:
done